Spec/mastermind_spec.rb

```diff
@@ -23,6 +23,100 @@
   end
 end
 
+RSpec.describe Code, "compares guess with secret_key" do
+  let(:code) { Code.new }
+  let(:colors) { ["r", "g", "o", "y", "b", "p"] }
+  let(:key) { ["r", "r", "r", "r"] }
+  let(:input) { ["r", "r", "r", "r"] }
+
+  it "returns number of red pins" do
+    expect(code.red_counter(key, input)).to eq [1, 1, 1, 1]
+  end
+end
+
+RSpec.describe Code, "compares guess with secret_key" do
+  let(:code) { Code.new }
+  let(:colors) { ["r", "g", "o", "y", "b", "p"] }
+  let(:key) { ["r", "r", "y", "b"] }
+  let(:input) { ["r", "r", "p", "o"] }
+
+  it "returns number of red pins" do
+    expect(code.red_counter(key, input)).to eq [1, 1]
+  end
+end
+
+RSpec.describe Code, "trims code" do
+  let(:code) { Code.new }
+  let(:key) { ["r", "g", "y", "r"] }
+  let(:input) { ["r", "o", "b", "r"] }
+
+  it "deletes input position if it is the same as key index" do
```

Comment:
For the same class described put all `it` cases within it and then have `contexts` or `describes` within it to give more detail on what it is doing. Keeps the specs tidy